src/nvim/terminal.c

```diff
@@ -496,37 +495,33 @@ static int terminal_execute(VimState *state, int key)
 void terminal_destroy(Terminal *term)
 {
   buf_T *buf = handle_get_buffer(term->buf_handle);
-  if (buf) {
-    term->buf_handle = 0;
-    buf->terminal = NULL;
-  }
+  assert(buf);
+  apply_autocmds(EVENT_TERMCLOSE, NULL, NULL, false, buf);
```

Comment:
The help says, this event is sent when the process exits, but if `terminal_destroy()` was called from `terminal_close()`,  the terminal job could still be running (for 4+ seconds, ignoring SIGHUP and SIGTERM, waiting for SIGKILL). If `terminal_destroy()` is called from the terminal job exit callback, the event is sent after the process
was killed.